### ISSUE TITLE
Cardinality aggregation of IP address field returns value_as_string formatted as an IP address 

### ISSUE THREAD

**Issue by evanv** [elastic/elasticsearch#21605]
**Elasticsearch version**: 2.4.1

**Plugins installed**: Shield, ES Restlog, Marvel agent, Kopf

**JVM version**: 1.8.0.60

**OS version**: Centos

**Description of the problem including expected versus actual behavior**:
If you run a cardinality aggregation on a IP address field, you get both a(n approximate) distinct count of IP addresses, as well as a `value_as_string` field formatted as an IP address

**Steps to reproduce**:
 1. execute the following query on an index that contains an IP field (cleverly named `ip`)

```
{
  "size": 0,
  "query": {
    "bool": {
      "must": [],
      "should": [],
      "must_not": [],
      "filter": [
        {
          "range": {
            "@timestamp": {
              "gte": "now-1d/d",
              "lt": "now/d"
            }
          }
        }
      ]
    }
  },
  "aggs": {
    "distinct_ips": {
      "cardinality": {
        "field": "ip"
      }
    }
  }
}
```

 2. You will get output that looks something like this:
```
{
    "_shards": {
        "failed": 0,
        "successful": 504,
        "total": 504
    },
    "aggregations": {
        "distinct_ips": {
            "value": 6448500,
            "value_as_string": "0.98.101.116"
        }
    },
    "hits": {
        "hits": [],
        "max_score": 0.0,
        "total": 79012098
    },
    "timed_out": false,
    "took": 1676
}
```
 

Not a huge deal by any means, but wanted to call it to your attention. Timestamps exhibit the same behavior. Strictly numeric fields (ints, floats, etc) do not.

**Comment by jpountz** [comment]
@colings86 Can you have a look?

**Comment by colings86** [comment]
It looks like this is already fixed in 5.0 as the `value_as_string` field is not used in the cardinality and value_count agg from 5.0. Since this is a benign bug and removing this in 2.4 could break some existing apps I think we should not fix this in 2.4. Since 5.0.0 is a major version it seems right that the breaking change is there. If anyone strongly disagrees with this please reopen this issue and comment to discuss this.
